modules/wechat_qrcode/src/wechat_qrcode.cpp

```diff
@@ -0,0 +1,210 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+//
+// Tencent is pleased to support the open source community by making WeChat QRCode available.
+// Copyright (C) 2020 THL A29 Limited, a Tencent company. All rights reserved.
+#include "precomp.hpp"
+#include "opencv2/wechat_qrcode.hpp"
+#include "decodermgr.hpp"
+#include "detector/align.hpp"
+#include "detector/ssd_detector.hpp"
+#include "opencv2/core.hpp"
+#include "opencv2/core/utils/filesystem.hpp"
+#include "scale/super_scale.hpp"
+#include "zxing/result.hpp"
+using cv::InputArray;
+namespace cv {
+namespace wechat_qrcode {
+class WeChatQRCode::Impl {
+public:
+    Impl() {}
+    ~Impl() {}
+    /**
+     * @brief detect QR codes from the given image
+     *
+     * @param img supports grayscale or color (BGR) image.
+     * @return vector<Mat> detected QR code bounding boxes.
+     */
+    std::vector<Mat> detect(const Mat& img);
+    /**
+     * @brief decode QR codes from detected points
+     *
+     * @param img supports grayscale or color (BGR) image.
+     * @param candidate_points detected points. we name it "candidate points" which means no
+     * all the qrcode can be decoded.
+     * @param points succussfully decoded qrcode with bounding box points.
+     * @return vector<string>
+     */
+    std::vector<std::string> decode(const Mat& img, std::vector<Mat>& candidate_points,
+                                    std::vector<Mat>& points);
+    int applyDetector(const Mat& img, std::vector<Mat>& points);
+    Mat cropObj(const Mat& img, const Mat& point, Align& aligner);
+    std::vector<float> getScaleList(const int width, const int height);
+    std::shared_ptr<SSDDetector> detector_;
+    std::shared_ptr<SuperScale> super_resolution_model_;
+    bool use_nn_detector_, use_nn_sr_;
+};
+
+WeChatQRCode::WeChatQRCode(const String& detector_prototxt_path,
+                           const String& detector_caffe_model_path,
+                           const String& super_resolution_prototxt_path,
+                           const String& super_resolution_caffe_model_path) {
+    p = makePtr<WeChatQRCode::Impl>();
+    if (!detector_caffe_model_path.empty() && !detector_prototxt_path.empty()) {
+        // initialize detector model (caffe)
+        p->use_nn_detector_ = true;
+        CV_CheckEQ(utils::fs::exists(detector_prototxt_path), true,
+                   "fail to find detector caffe prototxt file");
```

Comment:
`bool` values are not supported for now in `CV_Check` macros (see `ARMv7` builder warnings).
Lets use `CV_Assert()` instead for now:

```
CV_Assert(utils::fs::exists(detector_prototxt_path));
```

(similar cases are below)